### PR TITLE
:sparkles: Change load to async

### LINKED ISSUE #169: Make load function in token abstract class async
For some reason this method was defined as sync instead of async [here](https://github.com/SatelCreative/spylib/blob/be7c3960e790166fd745bdfae4bccf1c6ca62707/spylib/admin_api.py#L280) and [here](https://github.com/SatelCreative/spylib/blob/be7c3960e790166fd745bdfae4bccf1c6ca62707/spylib/admin_api.py#L296)

And maybe try to see if mypy could run on the test folder so we can pick up this kind of silly mistake.

### PR DESCRIPTION
**Pull request by ponty33**
Close #169 
- Change `def` to `async def` for the load functions

### CODE REVIEW

**Review by hillairet — APPROVED** [review]

**Review by hillairet — CHANGES_REQUESTED** [review]
Oh can you add a note in the docs actually? It's a breaking change!

**Comment by ponty33** [comment]
The example in the doc is already written as `async`...
https://github.com/SatelCreative/spylib/blob/f191886c597b0a26f4911484b22a149865b26e49/docs/admin-api.md?plain=1#L19
Any suggestion on where else I should write the reminder?

**Comment by hillairet** [comment]
> The example in the doc is already written as `async`...
> 
> https://github.com/SatelCreative/spylib/blob/f191886c597b0a26f4911484b22a149865b26e49/docs/admin-api.md?plain=1#L19
> 
> 
> Any suggestion on where else I should write the reminder?

At [this line](https://github.com/SatelCreative/spylib/blob/f191886c597b0a26f4911484b22a149865b26e49/docs/admin-api.md?plain=1#L36) you can something like:
*Note*: up to version 0.8.x, the load was sync instead of async.

Some sort of breaking change warning would be nice such as :boom:

**Review by lishanl — COMMENTED** [review]
mmm maybe it's more like 
upto `0.8.1` it can be `def load` or `async def load`, see https://github.com/SatelCreative/kitchenhub/blob/main/deliverect-integration-app/deliverect-integration-app/shopify/client.py#L15-L20
after `0.8.1` it can only be `async def load` 
?

**Comment by ponty33** [comment]
Prior to 0.8.1 should just be `sync`. I don't think you can override it with `async`

File: docs/admin-api.md [hunk 1]
```diff
@@ -42,26 +42,36 @@ from spylib.admin_api import OfflineTokenABC
...
-  async def load(cls, store_name: str):
-      pass
-      
-# Add the version to be used here
-OfflineToken.api_version: '2022-01'
+    @classmethod
+    async def load(cls, store_name: str):
+        pass
+
+    # For version <= 0.8.1, async is not available for load function
```
Comment (hillairet) [inline docs/admin-api.md]: They should be separate to make it clearer that it's in different case and classes.
````suggestion
```

:warning: For version <= 0.8.1, async is not available for load function
```python
class OfflineToken(OfflineTokenABC):
````

**Review by hillairet — CHANGES_REQUESTED** [review]
Emphasize the difference

File: docs/admin-api.md [hunk 2]
```diff
@@ -42,26 +42,34 @@ from spylib.admin_api import OfflineTokenABC
...
-      pass
-      
-# Add the version to be used here
-OfflineToken.api_version: '2022-01'
+    @classmethod
+    async def load(cls, store_name: str):
+        pass
+```
+
+**Warning:** For version <= 0.8.1, async is **not** available for load function
```
Comment (hillairet) [inline docs/admin-api.md]: The emoji is even better to catch the eye
```suggestion
:warning: **Warning:** For version <= 0.8.1, async is **not** available for load function
```

**Review by hillairet — APPROVED** [review]

File: docs/admin-api.md (hunk 2 above)
Comment (ponty33) [reply docs/admin-api.md]: Oh, didn't notice that's emoji, sure

**Pull request merged**